service/service.go

```diff
@@ -222,6 +234,10 @@ func (app *Application) GetExporters() map[configmodels.DataType]map[configmodel
 	return app.builtExporters.ToMapByDataType()
 }
 
+func (app *Application) GetResource() pdata.Resource {
```

Comment:
Can you provide an example of using this function in the metrics pipeline?
There will be either merge of the attributes from incoming request, host and [ResourceProcessor's generated resource](https://github.com/open-telemetry/opentelemetry-collector/blob/master/processor/resourceprocessor/resource_processor.go), or some preference order, I suppose.

Reply:
A simple example would be updating the resource processor component with a config option so that instead of providing static config you specify that you want the appended resource data to be auto-detected.

```yaml
processors:
    resource:
        auto: true # if auto = true, labels should be nil
```

We'd just need to store the auto-detected resource info on start up. The rest of the code to merge in this resource info would be unchanged.

```go
func (rmp *resourceMetricProcessor) Start(ctx context.Context, host component.Host) error {
    if rmp.cfg.AutoDetect {
        rmp.Resource = host.GetResource()
    }
}
```

In terms of merging resources from multiple sources, the existing convention as you mention above is to retain the earliest added resource information in the case of conflicts (as per what the resource processor is currently doing), so I'd expect this pattern to be followed everywhere.